### PR TITLE
Test improvements, add verifier-test.sh to ginkgo

### PR DESCRIPTION
**Pull request by joestringer**
Add the verifier test script to the regular ginkgo testsuite. If a datapath change is likely to cause verifier complexity issues, then this specific test should catch it.

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: test/config/config.go [hunk 1]
```diff
@@ -38,4 +39,6 @@ func (c *CiliumTestConfigType) ParseFlags() {
 		"Specify a custom command to fetch SSH configuration (eg: 'vagrant ssh-config')")
 	flag.BoolVar(&c.ShowCommands, "cilium.showCommands", false,
 		"Output which commands are ran to stdout")
+	flag.BoolVar(&c.GatherLogs, "cilium.gatherLogs", true,
```
Comment (ianvernon) [inline test/config/config.go]: Update documentation in `Documentation/contributing.rst` to reflect this new command.

File: test/runtime/verifier.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		res = vm.ExecWithSudo("rm -f /sys/fs/bpf/tc/globals/cilium*")
+		res.ExpectSuccess()
+	})
+
+	AfterAll(func() {
+		err := vm.RestartCilium()
+		Expect(err).Should(BeNil())
+	})
+
+	FIt("Runs verifier test script", func() {
```
Comment (ianvernon) [inline test/runtime/verifier.go]: FIt --> It ?

File: test/runtime/verifier.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+		By("Stopping Cilium")
+		res := vm.ExecWithSudo("systemctl stop cilium")
+		res.ExpectSuccess()
+		res = vm.ExecWithSudo("rm -f /sys/fs/bpf/tc/globals/cilium*")
+		res.ExpectSuccess()
+	})
+
+	AfterAll(func() {
+		err := vm.RestartCilium()
```
Comment (ianvernon) [inline test/runtime/verifier.go]: Add an `AfterEach` which gathers logs like the other runtime tests do.

File: test/runtime/ct.go [hunk 4]
```diff
@@ -507,7 +507,7 @@ var _ = Describe("DisabledRuntimeValidatedConntrackTable", func() {
 
 		By("Removing policy `l3-l4-policy-server-3`")
 		res = vm.PolicyDel("l3-l4-policy-server-3")
-		res.WasSuccessful()
+		res.ExpectSuccess()
```
Comment (ianvernon) [inline test/runtime/ct.go]: Good catch!

**Review by ianvernon — DISMISSED** [review]

File: test/config/config.go (hunk 1 above)
Comment (joestringer) [reply test/config/config.go]: done.

File: test/runtime/verifier.go (hunk 2 above)
Comment (joestringer) [reply test/runtime/verifier.go]: done.

File: test/runtime/verifier.go (hunk 3 above)
Comment (joestringer) [reply test/runtime/verifier.go]: done.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Unrelated test flake:

```
k8s-1.9.K8sValidatedPolicyTestAcrossNamespaces Policies Across Namespaces (from (k8s-1.9.xml))

Failing for the past 1 build (Since ￼#1640 )
Took 56 sec.
Stacktrace
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/Policies.go:744
Unable to connect between frontend-x8sqj and 10.97.215.210:80/
Expected
    <bool>: false
to be true
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/Policies.go:820
```

**Comment by joestringer** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hi, 

I'm happy with your changes, but you should also add `config.CiliumTestConfig.GatherLogs` in the kubernetes reportFailed. 

Regards

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

File: test/runtime/verifier.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	verifierTest = "RuntimeVerifier"
```
Comment (ianvernon) [inline test/runtime/verifier.go]: `RuntimeVerifier` --> `RuntimeValidatedVerifier`.

**Review by ianvernon — DISMISSED** [review]

File: test/runtime/verifier.go (hunk 5 above)
Comment (joestringer) [reply test/runtime/verifier.go]: What's the purpose of the "Validated" portion of the name at this point? I thought thit was just a temporary thing while we migrated from the old bash tests.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Looks like I just need to find the right target to invoke build of `bpf/` directory within Ginkgo VM, and that should fix up the CI for the new test.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Github messed up and added a bunch of unrelated owners because it thought one of the commits wasn't in master. Sorry about that.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Closing since most of the comments are related to commits that have been split into other PRs. I'll open a fresh PR to cover the specific ginkgo kernel verifier changes that remain here.

**Pull request closed without merging**